pkg/cmd/util/clientcmd/clientcache.go

```diff
@@ -0,0 +1,98 @@
+package clientcmd
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	"github.com/openshift/origin/pkg/client"
+)
+
+// clientCache caches previously loaded clients for reuse. This is largely
+// copied from upstream (because of typing) but reuses the negotiation logic.
+// TODO: Consolidate this entire concept with upstream's ClientCache.
+type clientCache struct {
```

Comment:
@smarterclayton objections to killing all this, just letting us negotiate purely against `/api`.  We only ever get back v1, which matches our code.  We get a config back, which monkey with enough to set `/oapi` and use that to build our origin clients?  All this duplicate caching and negotiating code makes me sad.